CDN-in-a-box enroller does not derefence string pointer

## I'm submitting a ...
-  bug report

## Traffic Control components affected ...
-  CDN in a Box

## Current behavior:

Executing tests on the cdn-in-a-box directory revealed a missing derefence for a string pointer

```bash
cd <trafficcontrol_directory>/infrastructure/cdn-in-a-box
go test -v ./...
```

Which will return

```bash
# github.com/apache/trafficcontrol/infrastructure/cdn-in-a-box/enroller
enroller/enroller.go:487:4: Infof format %s has arg s.Name of wrong type *string
```

## Expected behavior:

Since there are no test files, executing `go test -v ./...` should return:

```bash
?   	github.com/apache/trafficcontrol/infrastructure/cdn-in-a-box/enroller	[no test files]
```

## Minimal reproduction of the problem with instructions:

See steps under **_Current Behavior_**

## Anything else:

The enroller has the potential to hit a run time exception by attempting to format a *string instead of a string. The string variable will need to be checked if _nil_ and, if not, it will need to be derefenced:

```go
func enrollStatus(toSession *session, r io.Reader) error {
// ...
			log.Infof("status %s already exists\n", s.Name)
// ...
}
```

A fix for the code might read something akin to:

```go
func enrollStatus(toSession *session, r io.Reader) error {
// ...
			if s.Name == nil {
				temp := ""
				s.Name = &temp
			}
			log.Infof("status %s already exists\n", s.Name)
// ...
}
```

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

    https://apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

--- a/.github/workflows/go.vet.yml
+++ b/.github/workflows/go.vet.yml
@@ -51,6 +51,5 @@ jobs:
       run: |
         skip_analyzers=(
           -copylocks=false
-          -printf=false
         )
         go vet "${skip_analyzers[@]}" ./...

--- a/.golangci.yml
+++ b/.golangci.yml
@@ -34,4 +34,3 @@ linters-settings:
   govet:
     disable:
       - copylocks
-      - printf

--- a/cache-config/testing/ort-tests/tcdata/todb.go
+++ b/cache-config/testing/ort-tests/tcdata/todb.go
@@ -36,7 +36,7 @@ func (r *TCData) OpenConnection() (*sql.DB, error) {
 		sslStr = "disable"
 	}
 
-	fmt.Printf("User: %s, Password: len %s, Hostname: %s, DB: %s\n", r.Config.TrafficOpsDB.User, len(r.Config.TrafficOpsDB.Password), r.Config.TrafficOpsDB.Hostname, r.Config.TrafficOpsDB.Name)
+	fmt.Printf("User: %s, Password: len %d, Hostname: %s, DB: %s\n", r.Config.TrafficOpsDB.User, len(r.Config.TrafficOpsDB.Password), r.Config.TrafficOpsDB.Hostname, r.Config.TrafficOpsDB.Name)
 	db, err = sql.Open("postgres", fmt.Sprintf("postgres://%s:%s@%s/%s?sslmode=%s", r.Config.TrafficOpsDB.User, r.Config.TrafficOpsDB.Password, r.Config.TrafficOpsDB.Hostname, r.Config.TrafficOpsDB.Name, sslStr))
 
 	if err != nil {

--- a/experimental/traffic_ops_auth/traffic_ops_auth.go
+++ b/experimental/traffic_ops_auth/traffic_ops_auth.go
@@ -192,7 +192,7 @@ func HandleLogin(db *sql.DB, jwtSigningKey string, w http.ResponseWriter, r *htt
 	clms, ok := token.Claims.(jwt.MapClaims)
 	if !ok {
 		w.WriteHeader(http.StatusInternalServerError)
-		log.Printf("ERROR creating token for '%s': expected MapClaims in token, got %T", "token.Claims", token.Claims)
+		log.Printf("ERROR creating token for 'token.Claims': expected MapClaims in token, got %T", token.Claims)
 		return
 	}
 	clms["user"] = user

--- a/experimental/traffic_ops_auth/traffic_ops_auth.go
+++ b/experimental/traffic_ops_auth/traffic_ops_auth.go
@@ -192,7 +192,7 @@ func HandleLogin(db *sql.DB, jwtSigningKey string, w http.ResponseWriter, r *htt
 	clms, ok := token.Claims.(jwt.MapClaims)
 	if !ok {
 		w.WriteHeader(http.StatusInternalServerError)
-		log.Printf("%s ERROR creating token for '%s': expected MapClaims in token, got %T", token.Claims)
+		log.Printf("ERROR creating token for '%s': expected MapClaims in token, got %T", "token.Claims", token.Claims)
 		return
 	}
 	clms["user"] = user

--- a/experimental/traffic_ops_auth/traffic_ops_auth.go
+++ b/experimental/traffic_ops_auth/traffic_ops_auth.go
@@ -348,7 +348,7 @@ func main() {
 			break
 		}
 		if firstAllowedCreateRole == "" {
-			log.Printf("ERROR creating initial admin user: no roles are allowed to create users\n", err)
+			log.Printf("ERROR creating initial admin user: no roles are allowed to create users\n")
 			return
 		}
 		err := createUser(db, config.AdminUser, config.AdminPass, firstAllowedCreateRole)

--- a/infrastructure/cdn-in-a-box/enroller/enroller.go
+++ b/infrastructure/cdn-in-a-box/enroller/enroller.go
@@ -707,7 +707,7 @@ func enrollProfile(toSession *session, r io.Reader) error {
 			log.Infof("creating param %+v", param)
 			newAlerts, _, err := toSession.CreateParameter(param, client.RequestOptions{})
 			if err != nil {
-				log.Infof("can't create parameter %+v: %v", param, err, newAlerts.Alerts)
+				log.Infof("can't create parameter %+v: %s, %v", param, err, newAlerts.Alerts)
 				continue
 			}
 			eparam, err = toSession.getParameter(param, nil)

--- a/infrastructure/cdn-in-a-box/enroller/enroller.go
+++ b/infrastructure/cdn-in-a-box/enroller/enroller.go
@@ -540,7 +540,7 @@ func enrollStatus(toSession *session, r io.Reader) error {
 	if err != nil {
 		for _, alert := range alerts.Alerts {
 			if alert.Level == tc.ErrorLevel.String() && strings.Contains(alert.Text, "already exists") {
-				log.Infof("status %s already exists", s.Name)
+				log.Infof("status %s already exists", *s.Name)
 				return nil
 			}
 		}

